### PR TITLE
[BACK-53] Project Sentiment and Categorization

### PR DESCRIPTION
**Pull request by acreturs**
### Description: 
Implement description/ELI5 regarding the sentiment towards the project
### How I implemented: 
A file that contains all the chatGPT methods and a file that scrapes hackernews. 
### Other 
I am currently working on the categorization wihtin the same branch. Sorry. But I am still struggleing with the types  

In ScrapeRepoCategroies there is still a problem with the types (not eslint confirm) but since we need the functionality I will commit it know and modify it later. But I am just not able to to write the correct types for the complex response. There are prototypes within the types folder.

### CODE REVIEW

**Review by thomas-woehrle — CHANGES_REQUESTED** [review]
I did not take a thorough look at everything, because the duplicates are disturbing.
The hackernews sentiment stuff is good work!

Please address the requested changes or answer if you have another notion. 

Then I will do the second review and look at everything again

**Also: run prettier via npm run format -> your build might fail because of this**

File: packages/graphql_server/types/hackerNews.d.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        author: {
+          value: string
+          matchLevel: string
+          matchedWords: string[]
+        }
+      }
+    }[]
+  }
+  
+  export type HackerNewsStoriesResponseHitsArray = {
```
[the remark this answers is not in the document]
Comment (acreturs) [reply packages/graphql_server/types/hackerNews.d.ts:40]: no because otherwise eslint will throw errors

**Comment by thomas-woehrle** [comment]
run `npm run format` in the root folder `truffle-ai-backend`. 
Your build might be failing because of this

**Comment by pczern** [comment]
@acreturs 
> [warn] Code style issues found in 4 files. Forgot to run Prettier?

File: packages/graphql_server/src/api/openAIApi.ts
Comment (thomas-woehrle) [inline packages/graphql_server/src/api/openAIApi.ts:1]: Please add comments to the functions. One at the top of the function in the format specified on the Notion page (->LaFamiglia->Backend->Information->Documentation guidelines) and multiple smaller ones inside the code explaining what you are doing

File: packages/graphql_server/src/api/openAIApi.ts [hunk 2]
```diff
@@ -1,39 +1,167 @@
...
-    })
-  })
-  const data = (await response.json()) as OpenAIResponse
-  // this mostly becomes null when the free version of openai turbo api is used more often than 3 times per minute
-  return data.choices?.[0]?.message?.content ?? null
+import axios from 'axios'
+import { RequestBodyOpenAI, ResponseBodyOpenAi } from '../../types/openAIApi'
+export { getELI5FromReadMe }
+const model = 'gpt-3.5-turbo'
+const postURL = 'https://api.openai.com/v1/chat/completions'
```
Comment (thomas-woehrle) [inline packages/graphql_server/src/api/openAIApi.ts:5]: maybe rename this to `openAIapiUrl`. you can use f2 to rename a variable and its occurences in vscode

File: packages/graphql_server/src/scraping/hackerNewsScraper.ts
Comment (thomas-woehrle) [inline packages/graphql_server/src/scraping/hackerNewsScraper.ts:1]: rename to hackerNewsScraping.ts

File: packages/graphql_server/src/scraping/scrapeGithubCategories.ts
Comment (thomas-woehrle) [inline packages/graphql_server/src/scraping/scrapeGithubCategories.ts:1]: please move the function to the `githubApi.ts` file. It really only calls the api -> no scraping. 
After moving it to the githubApi.ts file you can delete this file

File: packages/graphql_server/types/hackerNews.d.ts
Comment (thomas-woehrle) [inline packages/graphql_server/types/hackerNews.d.ts:1]: rename this file to `hackerNewsScraper.d.ts` , even if its types are used somewhere else as well

**Review by thomas-woehrle — DISMISSED** [review]
Good improvements. 

Additionally to everything I said in my comments please add for each of the files that you added a file called `<filename>.test.ts` into the `/src/test` folder where you show possible invoactions of the methods. Look at the `dbUpdater.test.ts` file to see what i mean.

**Review by thomas-woehrle — CHANGES_REQUESTED** [review]
Sorry but you could please revise the test files. 

Right now all of the test functions are logging promises. 
Also please let the test functions take as parameters the parameters that the corresponding functions would take. I just want to write a single line of code to test it, because it takes a lot of time already to do reviews. 
F.e. the function testHackernewsScraping should take as parameter the parameter that searchHackernewsStory takes in and should log onto the console the result.

In the case of testHackerNewsSentiment, which also needs the readme you can decide on whether or not to pass arguments to the test function, since then you would have to get readme first. 

Write me if you have any questions.

**Review by pczern — APPROVED** [review]

**Comment by thomas-woehrle** [comment]
I changed some small things. @acreturs sorry I falsely wrote you to delete githubScraping. What I actually wanted was to just delete one funciton of it. I restored it to how it should be though

**Review by thomas-woehrle — APPROVED** [review]
